src/vs/workbench/services/languageRuntime/common/languageRuntimePlotClient.ts

```diff
@@ -274,7 +274,8 @@ export class PlotClientInstance extends Disposable implements IPositronPlotClien
 		// render request to be made multiple times, e.g. when the UI component
 		// is redrawn without changing the plot size.
 		if (this._lastRender &&
-			this._lastRender.size === sizeInt &&
+			this._lastRender.size?.height === sizeInt?.height &&
+			this._lastRender.size?.width === sizeInt?.width &&
```

Comment:
We were incorrectly checking the cached value here, comparing by pointer instead of value.

Reply:
Thanks for spotting this, that explains a lot of the spurious re-renders I've seen!